src/main/java/Duke.java

```diff
@@ -1,10 +1,212 @@
+                    }
+                } else {
+                    throw new DukeException("☹ OOPS!!! What is that? I don't know what is that? What is the meaning :-(");
+                }
+            } catch (DukeException de) {
+                System.out.println("____________________________________________________________");
+                System.out.println(de.getMessage());
+                System.out.println("____________________________________________________________");
+            }
+            saveTasksToFile();
+        }
+
+        System.out.println("____________________________________________________________");
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void processTask(String userInput, ArrayList<Task> tasks, boolean mark) {
+        int taskNumber;
+        try {
+            taskNumber = Integer.parseInt(userInput.split(" ")[1]);
+            if (taskNumber <= 0 || taskNumber > tasks.size()) {
+                throw new IndexOutOfBoundsException();
+            }
+
+            Task task = tasks.get(taskNumber - 1);
+            if (mark) {
+                task.markAsDone();
+                System.out.println("____________________________________________________________");
+                System.out.println("Nice! I've marked this task as done:");
+            } else {
+                task.unmarkAsDone();
+                System.out.println("____________________________________________________________");
+                System.out.println("OK, I've marked this task as not done yet:");
+            }
+            System.out.println("  [" + task.getStatusIcon() + "] " + task.description);
+            System.out.println("____________________________________________________________");
+        } catch (NumberFormatException e) {
+            System.out.println("Please provide a valid task number.");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("That task doesn't exist!");
+        }
+    }
+
+    public static void saveTasksToFile() {
+        try {
+            File file = new File(FILE_PATH);
+            if (!file.getParentFile().exists()) {
+                file.getParentFile().mkdirs(); // Create directories if they don't exist
+            }
+            file.createNewFile(); // Create the file if it doesn't exist
+            try (FileWriter writer = new FileWriter(file)) {
+                for (int i = 0; i < tasks.size(); i++) {
+                    writer.write(tasks.get(i).toFileString() + "\n");
+                }
+            }
+        } catch (IOException e) {
+            System.err.println("Error saving tasks to file: " + e.getMessage());
+        }
+    }
+
+    // Method to load tasks from a file
+    // Method to load tasks from a file
+    public static void loadTasksFromFile() {
```

Comment:
Use javadoc comments; in fact all non-trivial methods need well written header comments